new/mode-agnostic-data-collection.md

```diff
@@ -0,0 +1,554 @@
+
+## Motivation
+
+    As a Chef user who uses both Chef Client Mode and Chef Solo Mode (including the mode commonly known as "Chef Client Local Mode"),
+    I want to be able to collect data about my entire fleet regardless of their client operation type,
+    so that I may better understand the impacts of my changes and may better detect failures.
+
+## Definitions
+
+To eliminate ambiguity and confusion, the following terms are used throughout this RFC:
+
+ * **Chef**: the tool used to automate your system.
+ * **Chef Client Mode**: Chef configured in "client mode" where a Chef Server is used to provide Chef its resources and artifacts
+ * **Chef Solo Mode**: Chef configured in a mode that utilizes a local Chef Zero server. Formerly known as "Chef Client Local Mode" (run as `chef-client --local-mode`) before [RFC 031](https://github.com/chef/chef-rfc/blob/master/rfc031-replace-solo-with-local-mode.md) was implemented, available in Chef version 12.10.54 and later.
+ * **Chef Solo Legacy Mode**: Chef in the former Solo operational mode (run as `chef-solo`) before [RFC 031](https://github.com/chef/chef-rfc/blob/master/rfc031-replace-solo-with-local-mode.md) was implemented (in Chef versions earlier than 12.10.54), or Chef run as `chef-solo --legacy-mode` in Chef version 12.10.54 and later.
+
+## Specification
+
+Similar to how data is collected and reported for Chef Reporting, we expect to implement a new EventDispatch class/instance that collects data about the Chef run and reports it accordingly. Unlike Chef Reporting, the server that receives this data is **not** running on the Chef Server, allowing users to utilize this function whether they use Chef Server or not. No new data collection methods are expected to be implemented as a result of this change; this change serves to implement a generic way to report the collected data in a "webhook-like" fashion to a non-Chef-Server receiver.
+
+The implementation must work with Chef running in any mode:
+
+ * Chef Client Mode
+ * Chef Solo Mode
+ * Chef Solo Legacy Mode
+
+### Protocol and Authentication
+
+All payloads will be sent to the Data Collector server via HTTP POST to the URL specified in the `data_collector_server_url` configuration parameter. Users should be encouraged to use a TLS-protected endpoint.
+
+For the initial implementation, transmissions to the Data Collector server can optionally be authenticated with the use of a pre-shared token which will be sent in a HTTP header. Given that the receiver is not the Chef Server, existing methods of using a Chef `client` key to authenticate the request are unavailable.
```

Comment:
What is the goal of this token?  Assuming it is a single shared token, you don't get much in the way of authentication (since any node could pretend to be any other) but you do at least get a bit of authorization (only token holders can make requests).  If you plan on having a token-per-node then some public key encryption feels like it would make more sense since you already have to manage N tokens.